### DISCUSSION TITLE
Feature Request: Grouping of Audiobooks

### DISCUSSION THREAD

**Discussion by u42671** [PaulWoitaschek/Voice#1314] [Ideas]
If you have hundreds of audio books, belonging to different audiobooks series scrolling becomes quite cumbersome when looking for a specific audiobook.
Especially if the scrolling position is lost when navigating to another screen and back and you have to start scrolling all over again.

Therefore it would be nice if MaterialAudiobookPlayer would support some kind of grouping mechanism where you can assign audiobooks which belong together to the same group.

Doesn't need to be something fancy.

Proposal:
- Allow creation of groups in AudioBookPlayer. Just a flat list, no hierarchy.
- There is a default group named "Unassigned"
- The AudioBookPlayer startscreen shows the list of defined groups including a number how many audiobooks are in a group.
- Once you click on such a group you see the list of audiobooks assigned to this group.
- When adding a new audiobook folder, you can specify to which group the audiobooks in this folder shall be assigned. If nothing is chosen, default is the group "Unassigned".

**Comment by kgraefe** [comment]
This would add some extra effort for those who don't use this feature as they would have to click "Unassigned" before they get to their audiobooks.  Therefore,
-  the "Unassigned" group should either be opened at the start screen by default when there are no other groups, or
- unassigned audiobooks should be at the start screen besides the other groups (without an extra "Unassigned" group)

**Comment by nioncode** [comment]
Related to #487.
Showing unassigned books next to the created groups sounds like the best approach.
Also, when there is only a single group (and no remaining books that are unassigned), this group should be opened automatically on startup. Thus, users who don't use groups don't even notice the feature.

It would be great if we could automatically assign groups based on the directory structure, as described in #487:

> /AudioBooks/1984/1984.mp3
> /AudioBooks/Dark Matter/Dark Matter.mp3
> /AudioBooks/Enders Saga/Enders Game/Enders Game.mp3
> /AudioBooks/Enders Saga/Speaker for the Dead/Speaker for the Dead.mp3

This would show '1984' and 'Dark Matter' as single books next to the automatically created group 'Enders Saga' which contains 'Enders Game' and 'Speaker for the Dead'.

**Comment by hakarune** [comment]
I'd like to group all the Lord if the rings books up and other series too

**Comment by leafcutterant** [comment]
I have the similar problem and need and would like to ask for a solution to it.

This was my case:

audiobooks/authorX/book1.mp3
audiobooks/authorX/book2.mp3
audiobooks/authorX/book3/chapter1.mp3
audiobooks/authorX/book3/chapter2.mp3
...
audiobooks/authorX/book3/chapter10.mp3

What Material Player did was displaying only book3 from authorX, and adding book1 and book2 as chapters to the _end_ of book3's chapter list. I couldn't for the love of life find out how to play book1 and book2 until I spent time sifting through the closed issues.

Imuli's strategy from #545 didn't solve t for me, it doesn't allow adding a folder if it's a subfolder of an already added folder.

I know this is a very difficult problem to solve. But until it's solved, you could allow adding subfolders. Or do something based on ID3 tag recognition so if we tag the files properly, it will display properly.

Enormous thanks for the good work!

**Comment by averageunicorn** [comment]
I came here because of the books-of-a-series-in-subfolders situation exactly.

I'd expect to have my folder structure reflected: Only adding my "Audiobooks" folder (as a folder with multiple books), seeing its subfolder "Harry Potter" in the overview, and when I tap that, a list of the 7 folders/books inside my Harry Potter folder. (Instead of everything inside "Harry Potter" showing up as one book, which can only be navigated via the chapter list.)

For short series with well-named files, it's manageable as it is. But, well, for others it is not much fun.

So, I'm another one who would be super happy about a solution for this! It's currently the only thing that makes it necessary for me to use other apps.

**Comment by thw0rted** [comment]
I'm pretty sure the discussion here is covered in #809 , so this can probably be closed.

**Comment by mesonium** [comment]
Somewhat related: https://github.com/PaulWoitaschek/Voice/discussions/1356
